prov/rxm/src/rxm_ep.c

```diff
@@ -833,16 +821,16 @@ rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
 static ssize_t rxm_ep_sendmsg(struct fid_ep *ep_fid, const struct fi_msg *msg,
 			      uint64_t flags)
 {
-
 	return rxm_sendmsg(ep_fid, msg, flags, 0, ofi_op_msg, FI_MSG);
 }
 
 static ssize_t rxm_ep_send(struct fid_ep *ep_fid, const void *buf, size_t len,
 			   void *desc, fi_addr_t dest_addr, void *context)
 {
-	struct iovec iov;
-	iov.iov_base = (void *) buf;
-	iov.iov_len = len;
+	struct iovec iov = {
+		.iov_base = (void *)buf,
```

Comment:
If you declare iov as 'const struct iovec', can we remove the buf cast to (void*) that is used to toss away the const from the buf definition?  This comment applies everywhere this cast is done.

Reply:
I've checked, this produces warning anyway, for instance:
```
prov/rxm/src/rxm_ep.c:831:3: warning: initialization discards 'const' qualifier from pointer target type [enabled by default]
   .iov_base = buf,
```

Reply:
Thanks for checking.